core/Segment.php

```diff
@@ -433,7 +433,7 @@ protected function getCleanedExpression($expression)
         // This is required to ensure segments like actionUrl!@value really do not include any visit having an action containing `value`
         if ($this->doesSegmentNeedSubquery($matchType, $name)) {
             $operator = $this->getInvertedOperatorForSubQuery($matchType);
-            $stringSegment = $name . $operator . $value;
+            $stringSegment = $name . $operator . preg_replace('/((?<!\\\)[,;])/', '\\\$1', $value);
```

Comment:
There are other variants that at least pass the added tests.

Encoding only the value:

```suggestion
            $stringSegment = $name . $operator . urlencode($value);
```

Encoding the complete segment:

```suggestion
            $stringSegment = urlencode($name . $operator . $value);
```

Does anyone find a case for one of the three solutions to be the "best" here?

Reply:
I think we should only either escape or encode the value only, not the whole segment. It's easier to debug/work with in the code when you can see the 'control'/functionally-significant characters, in this case the `,` and the `;` that we use as operators for chaining the segments.
And from the two options the `urlencode` is easier to comprehend vs looking at the preg_replace regex, so it would be my choice.

⚡ Code for future readability, not for current cool factor.

Reply:
Moved the scaping to a seperate function as mentioned (with a description of what is done to turn down the magic factor). We can always update the chosen way once we have information if any other option behaves differently/better.